### PR TITLE
feat: integrate logic for handling multiple charge stations

### PR DESCRIPTION
**Pull request by CodeBear801**
issue: https://github.com/Telenav/osrm-backend/issues/143

- Enable logic for `generateSolutions4MultipleCharge`
- Implement logic for converting from IngernalSolution to External response
- Move function of `GenerateFakeResponse` from `handler` to `oasis/mock_response`
- Move function of `generateOASISResponse4NoChargeNeeded` from `handler` to `has_enough_energy`
- Move function of `RequestRoute4InputOrigDest` from `handler` to `osrmhelper`

## Tasklist

 - [x] update relevant code
 - [ ] review

### CODE REVIEW

File: integration/oasis/reachable_by_multiple_charge.go [hunk 1]
```diff
@@ -1,24 +1,52 @@
...
-	// 	sol := stationgraph.NewStationGraph(c, oasisReq.CurrRange, oasisReq.MaxRange,
-	// 		chargingstrategy.NewFakeChargingStrategy(oasisReq.MaxRange)).GenerateChargeSolutions()
-	// }
+// @todo: handle negative situation
+func generateSolutions4MultipleCharge(w http.ResponseWriter, oasisReq *oasis.Request, routeResp *route.Response, oc *osrmconnector.OSRMConnector, sc *searchconnector.TNSearchConnector) {
+	solutions := generateSolutionsWithEarlistArrival(oasisReq, routeResp, oc, sc)
+
+	w.WriteHeader(http.StatusOK)
+	r := new(oasis.Response)
+	r.Code = "200"
```
Comment (wangyoucao577) [inline integration/oasis/reachable_by_multiple_charge.go]: Nit: have seen several this hardcode, use `const` instead.

File: integration/oasis/solution/solution.go [hunk 2]
```diff
@@ -24,3 +29,38 @@ type Location struct {
 	Lat float64
 	Lon float64
 }
+
+// Convert2ExternalSolution convert internal solution format to external format defined in pkg/api/oasis/response
+func (sol *Solution) Convert2ExternalSolution() *oasis.Solution {
```
Comment (wangyoucao577) [inline integration/oasis/solution/solution.go]: Where is the "oasis.Solution"? I went through codes but didn't find it.

Comment (CodeBear801) [reply integration/oasis/solution/solution.go]: https://github.com/Telenav/osrm-backend/blob/0cf096845aa6067be8625ee4abb671af56c5abc5/integration/pkg/api/oasis/response.go#L13

File: integration/oasis/reachable_by_multiple_charge.go (hunk 1 above)
Comment (CodeBear801) [reply integration/oasis/reachable_by_multiple_charge.go]: 👍 Yes, better to use const value.

File: integration/oasis/solution/solution.go (hunk 2 above)
Comment (wangyoucao577) [reply integration/oasis/solution/solution.go]: Oh! Got it! Currently it's ok to me. Let's discuss it later, same question on osrm api/impl too.

**Pull request merged**